numpy/f2py/capi_maps.py

```diff
@@ -241,10 +241,7 @@ def getctype(var):
     """
     ctype = 'void'
     if isfunction(var):
-        if 'result' in var:
-            a = var['result']
-        else:
-            a = var['name']
+        a = var.get('result', var['name'])
```

Comment:
This does the exact same amount of work as the old code (two lookups), except that it assumes that `name` exists (ok, the python `if` above might slow things down if we are micro-optimizing). If someone else likes this, OK. But I think I prefer the old code.

Reply:
@seberg, it looks like both versions assume that `name` is in `var`.  What am I missing?  The change is four lines of code to an equivalent 1.  What's not to like?

Reply:
The exact same pattern occurs (in this edited file) at lines 286, 377, 652, and 777.  If we make this change, we might as well change those, too.

Reply:
@seberg, nevermind, now I see your point.

Reply:
If we're actually trying to save lookups here, the way to achieve that is a try/except.

Reply:
AFAIK get internally uses try catch

Reply:
Hypothetically, the new code could raise an exception where the old code doesn't.  This could happen if `var` has been created such that *either* `result` or `name` is a key, but not both.  The old code works, but the new shorter code raises an exception.  I have no idea if that is the actual case here; I suspect not.  Still, I'm now inclined to agree with @seberg: let it be.

Reply:
Well, I admit its a lot shorter, so maybe it is just a pattern I am not used to, unless python now does magic here though it is not faster. As for try/except, that is faster, but only if you know the default is taken almost always.